File: data/EXP01/python/2266/2266.output_1.py
# @lc app=leetcode id=2266 lang=python3
class Solution:
    def countTexts(self, pressedKeys: str) -> int:
        mappings = ['', '', 'abc', 'def', 'ghi', 'jkl', 'mno', 'pqrs', 'tuv', 'wxyz']
        n = len(pressedKeys)
        dp = [0] * (n + 1)
        dp[0] = 1

        for i in range(n):
            digit = int(pressedKeys[i])
            letters = mappings[digit]
            for j in range(1, len(letters)+1):
                dp[i+j] = (dp[i+j] + dp[i]) % (10**9 + 7)
        
        return dp[n]

File: data/EXP01/golang_no_system_var/2266/2266.output_1.go
func countTexts(pressedKeys string) int {
	dp := make([]int, len(pressedKeys)+1)
	dp[0] = 1

	for i := 1; i <= len(pressedKeys); i++ {
		for j := 2; j <= 9; j++ {
			if pressedKeys[i-1] == byte(j+'0') {
				dp[i] = (dp[i] + dp[i-1]) % (1e9 + 7)
			}
		}
	}

	return dp[len(pressedKeys)]
}
